staging/src/k8s.io/apiserver/pkg/storage/value/transformer.go

```diff
@@ -189,7 +192,11 @@ func (t *prefixTransformers) TransformFromStorage(data []byte, context Context)
 	if err := errors.Reduce(errors.NewAggregate(errs)); err != nil {
 		return nil, false, err
 	}
-	RecordTransformation("from_storage", "unknown", start, t.err)
+	RecordTransformation(
+		"from_storage",
+		"unknown",
+		start,
+		status.Error(codes.Internal, t.err.Error()))
```

Comment:
This is basically "something is wrong with how you configured encryption at rest" right?

Is there a good code for "config error?"  The `NotFound` and `DataLoss` codes vaguely match but seem like a stretch.

Reply:
I am struggling with finding the appropriate status code myself, though I think Internal is better then Unknown since we know precisely what caused the error.
DataLoss is probably not accurate since (by correcting the config) the data could be recovered.
NotFound would work as well, though I think that some KMSes uses this code to indicate that the key no longer exists.

Another options:
1. Attach details to the status ([example](https://jbrandhorst.com/post/grpc-errors/)).
2. Create a custom error for this condition
Then in the metric's logic either check for the error's type or examine status' details and emit a custom label (prefixMismatch) for this condition.

IMO, the combination of "prefix=unknown" and Status=Internal should be enough to define an alert for this condition.

Reply:
> IMO, the combination of "prefix=unknown" and Status=Internal should be enough to define an alert for this condition.

👍